templates/common/fs.go

```diff
@@ -124,6 +124,23 @@ type CopyParams struct {
 	// source, to allow customization of the copy operation on a per-file basis.
 	Visitor CopyVisitor
 
+	// Return [SymlinkForbiddenError] if any symlinks are found when copying.
+	//
+	// Why are symlinks risky in a template?
+	//
+	//  - A symlink could point outside the template directory, resulting in
+	//    unintended exfiltration of user data into the template output
+	//  - Users might be surprised if a template outputs a symlink, which
+	//    could point to any location outside the render destination dir.
+	//
+	// In general, symlinks could cause all kinds of surprising behavior. We're
+	// trying to provide a bit of protection for template users against
+	// malicious or overzealous template authors.
+	//
+	// We could be more permissive in the future and allow symlinks to exist in
+	// the repo as long as they're not involved with the template.
+	ForbidSymlinks bool
```

Comment:
Will we default to allow symlinks?

Reply:
Hmm good point, I'll invert `ForbidSymlinks` to `AllowSymlinks` so the default is more safe.

Reply:
I ended up just hardcoding the symlink check. I don't think we ever want to allow symlinks, at least for now. We can make it configurable later, if needed.